src/main/java/org/jabref/logic/importer/fetcher/ScholarArchiveFetcher.java

```diff
@@ -0,0 +1,167 @@
+package org.jabref.logic.importer.fetcher;
+
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+import kong.unirest.json.JSONObject;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.lucene.queryparser.flexible.core.nodes.QueryNode;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.PagedSearchBasedParserFetcher;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.importer.fetcher.transformers.ScholarArchiveQueryTransformer;
+import org.jabref.logic.util.OS;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ScholarArchiveFetcher implements PagedSearchBasedParserFetcher {
+
+    // Define a constant for the fetcher name.
+    public static final String FETCHER_NAME = "ScholarArchive";
+
+    // Initialize the logger for this class.
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScholarArchiveFetcher.class);
+
+    // Define the API URL for ScholarArchive.
+    private static final String API_URL = "https://scholar.archive.org/search";
+
+    @Override
+    /**
+     * Gets the query URL.
+     *
+     * @param luceneQuery the search query
+     * @param pageNumber  the number of the page indexed from 0
+     * @return URL
+     */
+    public URL getURLForQuery(QueryNode luceneQuery, int pageNumber) throws URISyntaxException, MalformedURLException, FetcherException {
+        URIBuilder uriBuilder = new URIBuilder(API_URL);
+
+        // Add search query parameter to the URL.
+        uriBuilder.addParameter("q", new ScholarArchiveQueryTransformer().transformLuceneQuery(luceneQuery).orElse(""));
+
+        // Add page number and page size parameters to the URL.
+        uriBuilder.addParameter("from", String.valueOf(getPageSize() * pageNumber));
+        uriBuilder.addParameter("size", String.valueOf(getPageSize()));
+
+        // Specify the response format as JSON.
+        uriBuilder.addParameter("format", "json");
+
+        // Build the URL.
+        return uriBuilder.build().toURL();
+    }
+
+    @Override
+    public Parser getParser() {
+        return inputStream -> {
+            // Read the API response into a string.
+            String response = new BufferedReader(new InputStreamReader(inputStream)).lines().collect(Collectors.joining(OS.NEWLINE));
```

Comment:
See CrossRef for an example             JSONObject response = JsonReader.toJsonObject(inputStream);